### ISSUE TITLE
TOPIC with empty Trailing message is wrongly treated as equivalent to TOPIC without Trailing message

### ISSUE THREAD

**Issue by stapelberg** [sorcix/irc#2]
According to https://tools.ietf.org/html/rfc2812#section-3.2.4, these two messages are different:

```
   TOPIC #test :                   ; Command to clear the topic on
                                   #test.

   TOPIC #test                     ; Command to check the topic for
                                   #test.
```

However, this code prints “broken”:

```go
package main

import (
    "fmt"

    "github.com/sorcix/irc"
)

func main() {
    m1 := irc.ParseMessage("TOPIC #foo :")
    m2 := irc.ParseMessage("TOPIC #foo")
    if string(m1.Bytes()) == string(m2.Bytes()) {
        fmt.Println("broken")
    }
}
```

I think we should add a HasTrailing bool to `irc.Message` so that these two messages can be distinguished. What do you think?

**Comment by sorcix** [comment]
You're right. This is an issue with a clear usecase.

**Comment by sorcix** [comment]
I'm wondering if something like HasTrailing is the best solution here. That allows having a Message struct with both a non-empty Trailing string and HasTrailing set to false, which is ugly and hard to understand.

**Comment by stapelberg** [comment]
I think `EmptyTrailing` makes more sense:

When the user creates a message, EmptyTrailing will be false by default, meaning that Trailing == "" will be encoded as "".

When the user creates a message with EmptyTrailing == true, Trailing == "" will be encoded as ":".

ParseMessage sets EmptyTrailing to true if and only if the message ends in ":".

**Comment by sorcix** [comment]
Fix implemented in c9a40650fe4660b6a03f2b0eccc615dad05a087d.

It's on a separate branch as I want to be absolutely sure before pushing anything to master, due to the way Go handles imports. Can you check if my change works as you'd expect?

**Comment by stapelberg** [comment]
I can confirm that the branch “emptytrailing” is working for me for this feature. Thanks for the quick fix!

(side note: you may want to add “fixes #2” to the commit message so that github auto-closes this bug)
